### ISSUE TITLE
CSS: Unsupported selector type ?!

### ISSUE THREAD

**Issue by kutoman** [codenameone/CodenameOne#2748]
I'm suddenly getting the following error when running the simulator:

```

CSS> java.lang.RuntimeException: Unsupported selector type 10
CSS> 	at com.codename1.designer.css.CSSTheme.getElementForSelector(CSSTheme.java:5746)
CSS> 	at com.codename1.designer.css.CSSTheme.apply(CSSTheme.java:5775)
CSS> 	at com.codename1.designer.css.CSSTheme$2.property(CSSTheme.java:6236)
CSS> 	at org.w3c.flute.parser.Parser.declaration(Parser.java:2090)
CSS> 	at org.w3c.flute.parser.Parser.styleRule(Parser.java:1405)
CSS> 	at org.w3c.flute.parser.Parser.afterImportDeclaration(Parser.java:561)
CSS> 	at org.w3c.flute.parser.Parser.parserUnit(Parser.java:498)
CSS> 	at org.w3c.flute.parser.Parser.parseStyleSheet(Parser.java:107)
CSS> 	at com.codename1.designer.css.CSSTheme.load(CSSTheme.java:6244)
CSS> 	at com.codename1.designer.css.CN1CSSCLI.compile(CN1CSSCLI.java:283)
CSS> 	at com.codename1.designer.css.CN1CSSCLI.main(CN1CSSCLI.java:222)
CSS> 	at com.codename1.designer.ResourceEditorApp.main(ResourceEditorApp.java:528)
CSS> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
CSS> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
CSS> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
CSS> 	at java.lang.reflect.Method.invoke(Method.java:498)
CSS> 	at com.codename1.impl.javase.JavaFXLoader.runWithJavaFX(JavaFXLoader.java:263)
CSS> 	at com.codename1.impl.javase.JavaFXLoader.main(JavaFXLoader.java:323)
CSS> 	at com.codename1.designer.ResourceEditorApp.main(ResourceEditorApp.java:259)
```

**Comment by kutoman** [comment]
I've found the cause as there was some string in the css out of place. I think it would be helpful if the line number to the css file would be mentioned in the error

**Comment by shannah** [comment]
I have made some improvements to the error reporting in the CSS compiler.  Now, the error that you posted here will include details about which selector was being processed, and what line number of the CSS file it occurred on.

Some classes of errors won't include a CSS line number because they occur during the .res file building phase. In these cases I have also improved error reporting so that it will show you which UIID and property were being processed at the time of the error.

This will be included in the next update.
